xarray/backends/file_manager.py

```diff
@@ -231,6 +231,11 @@ def close(self, needs_lock=True):
             default = None
             file = self._cache.pop(self._key, default)
             if file is not None:
+                # Check for string so we do not have to import it
+                if str(type(file)) == "<class 'netCDF4._netCDF4.Dataset'>":
+                    with NETCDF4_PYTHON_LOCK:
+                        file.close()
+                        return
```

Comment:
Can we add a parameter to FileManager (and also probably PickleableFileManager) in a way that allows this logic to stay only in the backend class?

I'm imagining adding a parameter `closer`, which is used to close files like `closer(file)`.

Reply:
Could also call this `close` but elsewhere in file_manager this is used for functions with a signature like `close()`, which could be a bit ambiguous.

Reply:
I noticed when can just store the lock in the FileManager, then that does not require significant changes, including making it self-contained, without needing a `closer` function